### PR TITLE
onnx 1.14.0 (new formula)

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
@ilya-lavrenov thanks for the PR, you can try using `brew style --fix onnx` to fix the style issues. Also, can you use `onnx 1.14.0 (new formula)` to amend your commit message?

File: Formula/onnx.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  desc "Open standard for machine learning interoperability"
+  homepage "https://onnx.ai"
+  url "https://github.com/onnx/onnx/archive/refs/tags/v1.14.0.tar.gz"
+  sha256 "1b02ad523f79d83f9678c749d5a3f63f0bcd0934550d5e0d7b895f9a29320003"
+  license "Apache-2.0"
+  head "https://github.com/onnx/onnx.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "pybind11" => :build
+  depends_on "python@3.10" => [:build, :test]
```
Comment (carlocab) [inline Formula/onnx.rb:11]:
```suggestion
```
I suggest using only `python@3.11` unless we know we need `python@3.10` too.

File: Formula/onnx.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        .select { |f| f.name.match?(/^python@\d\.\d+$/) }
+        .sort_by(&:version)
+        .map { |f| f.opt_libexec/"bin/python" }
+  end
+
+  def install
+    # Remove git cloned 3rd party to make sure formula dependencies are used
+    dependencies = %w[third_party/benchmark
+                      third_party/pybind11]
+    dependencies.each { |d| (buildpath/d).rmtree }
```
Comment (carlocab) [inline Formula/onnx.rb:27]:
```suggestion
    (buildpath/"third_party").children.map(&:rmtree)
```

File: Formula/onnx.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    cmake_args = std_cmake_args + %w[
+      -DONNX_ML=ON
+      -DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *cmake_args
```
Comment (carlocab) [inline Formula/onnx.rb:34]:
```suggestion
    cmake_args = %w[
      -DONNX_ML=ON
      -DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON
    ]

    system "cmake", "-S", ".", "-B", "build", *cmake_args, *std_cmake_args
```
Pass `std_cmake_args` last.

File: Formula/onnx.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    system "cmake", "--install", "build"
+
+    pythons.each do |python|
+      ENV.append "ONNX_ML", "1"
+      ENV.append "CMAKE_ARGS", "-DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON"
+      system python, *Language::Python.setup_install_args(libexec, python)
+
+      site_packages = Language::Python.site_packages(python)
+      pth_contents = "import site; site.addsitedir('#{libexec/site_packages}')\n"
+      (prefix/site_packages/"homebrew-onnx.pth").write pth_contents
```
Comment (carlocab) [inline Formula/onnx.rb:45]: Why not
```suggestion
      system python, *Language::Python.setup_install_args(prefix, python)
```
?

File: Formula/onnx.rb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      #include <iostream>
+      #include <onnx/common/version.h>
+      int main()
+      {
+        std::cout << ONNX_NAMESPACE::LAST_RELEASE_VERSION;
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "test.cpp", "-std=c++17",
+           "-I#{include}", "-L#{lib}", "-lonnx", "-o", "test"
```
Comment (carlocab) [inline Formula/onnx.rb:60]: This seems short enough to fit a single line.
```suggestion
    system ENV.cxx, "test.cpp", "-std=c++17", "-I#{include}", "-L#{lib}", "-lonnx", "-o", "test"
```

File: Formula/onnx.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      -DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    pythons.each do |python|
+      ENV.append "ONNX_ML", "1"
+      ENV.append "CMAKE_ARGS", "-DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON"
```
Comment (carlocab) [inline Formula/onnx.rb:40]:
```suggestion
      ENV["ONNX_ML"] = "1"
      ENV["CMAKE_ARGS"] = "-DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON"
```

**Pull request closed without merging**